db/mongo/mongoClient_test.go

```diff
@@ -0,0 +1,214 @@
+			_, err := m.AddStudent(tt.args.student)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("AddStudent() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+		})
+	}
+}
+
+func Test_mongoStd_UpdateStudent(t *testing.T) {
+	os.Setenv("DB_PORT", "27017")
+	os.Setenv("DB_HOST", "student-portal-mongo-db")
+
+	type args struct {
+		student *models.Student
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "success - update records",
+			args: args{student: &models.Student{
+				Name:  "Ateeq",
+				Age:   21,
+				Level: "Bachelor",
+				Phone: "03005478963",
+			},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			m, _ := NewMongoClient(db.Option{})
+			if err := m.UpdateStudent(tt.args.student); (err != nil) != tt.wantErr {
+				t.Errorf("UpdateStudent() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func Test_mongoStd_GetStudentByID(t *testing.T) {
+	os.Setenv("DB_PORT", "27017")
+	os.Setenv("DB_HOST", "student-portal-mongo-db")
+
+	m, _ := NewMongoClient(db.Option{})
+	student := &models.Student{
+		Name:  "Ali",
+		Age:   22,
+		Level: "Bachelor",
+		Phone: "03005124896",
+	}
+	_, _ = m.AddStudent(student)
+	type args struct {
+		id string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *models.Student
+		wantErr bool
+	}{
+		{
+			name:    "success - get student by ID",
+			args:    args{id: student.ID},
+			want:    student,
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := m.GetStudentByID(tt.args.id)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("GetStudentByID() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("GetStudentByID() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_mongoStd_RemoveStudentByID(t *testing.T) {
+	os.Setenv("DB_PORT", "27017")
+	os.Setenv("DB_HOST", "student-portal-mongo-db")
+
+	m, _ := NewMongoClient(db.Option{})
+	student := &models.Student{
+		Name:  "Usman",
+		Age:   22,
+		Level: "Bachelor",
+		Phone: "03005124896",
+	}
+	_, _ = m.AddStudent(student)
+	type args struct {
+		id string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *models.Student
+		wantErr bool
+	}{
+		{
+			name:    "success - delete student by ID",
+			args:    args{id: student.ID},
+			want:	 student,
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := m.RemoveStudentByID(tt.args.id)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("RemoveStudentByID() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+		})
+	}
+}
+
+func Test_mongoStd_ListStudent(t *testing.T) {
+	os.Setenv("DB_PORT", "27017")
+	os.Setenv("DB_HOST", "student-portal-mongo-db")
+	m, _ := NewMongoClient(db.Option{})
+	student := []models.Student{{
```

Comment:
Instead of list.. create two separate students. And add two test cases that list 1 item each according to the filter passed.

And want would become want: []*models.Student{student1}

Or student2 based filter.